src/Cli/dotnet/Commands/Solution/Add/SolutionAddCommand.cs

```diff
@@ -44,115 +51,128 @@ public override int Execute()
+
+        if (string.IsNullOrEmpty(_solutionFolderPath))
         {
+            // Generate the solution folder path based on the project path
+            relativeSolutionFolderPath = Path.GetDirectoryName(relativeProjectPath);
+
+            // If the project is in a folder with the same name as the project, we need to go up one level
+            if (relativeSolutionFolderPath.Split(Path.DirectorySeparatorChar).LastOrDefault() == Path.GetFileNameWithoutExtension(relativeProjectPath))
             {
-                if (ex is SolutionException || ex.InnerException is SolutionException)
-                {
-                    throw new GracefulException(CliStrings.InvalidSolutionFormatString, solutionFileFullPath, ex.Message);
-                }
-                throw new GracefulException(ex.Message, ex);
+                relativeSolutionFolderPath = Path.Combine([.. relativeSolutionFolderPath.Split(Path.DirectorySeparatorChar).SkipLast(1)]);
+            }
+
+            // If the generated path is outside the solution directory, we need to set it to empty
+            if (!IsSolutionFolderPathInDirectoryScope(relativeSolutionFolderPath))
+            {
+                relativeSolutionFolderPath = string.Empty;
             }
         }
+        else
+        {
+            // Use the provided solution folder path
+            relativeSolutionFolderPath = _solutionFolderPath;
+        }
+
+        return string.IsNullOrEmpty(relativeSolutionFolderPath)
+            ? null
+            : solution.AddFolder(GetSolutionFolderPathWithForwardSlashes(relativeSolutionFolderPath));
     }
 
-    private async Task AddProjectsToSolutionAsync(string solutionFileFullPath, IEnumerable<string> projectPaths, CancellationToken cancellationToken)
+    private async Task AddProjectsToSolutionAsync(IEnumerable<string> projectPaths, CancellationToken cancellationToken)
     {
-        SolutionModel solution = SlnFileFactory.CreateFromFileOrDirectory(solutionFileFullPath);
+        SolutionModel solution = SlnFileFactory.CreateFromFileOrDirectory(_solutionFileFullPath);
         ISolutionSerializer serializer = solution.SerializerExtension.Serializer;
+
         // set UTF8 BOM encoding for .sln
         if (serializer is ISolutionSerializer<SlnV12SerializerSettings> v12Serializer)
         {
             solution.SerializerExtension = v12Serializer.CreateModelExtension(new()
             {
                 Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true)
             });
+
             // Set default configurations and platforms for sln file
-            foreach (var platform in _defaultPlatforms)
-            {
-                solution.AddPlatform(platform);
-            }
-            foreach (var buildType in _defaultBuildTypes)
-            {
-                solution.AddBuildType(buildType);
-            }
+            SlnFileFactory.DefaultPlatforms.ToList().ForEach(solution.AddPlatform);
```

Comment:
I think I commented on this at some point, but I prefer the for loops

Reply:
Yes you are very right, completely forgot about that one sorry